slogr/slogr.go

```diff
@@ -0,0 +1,46 @@
+//go:build go1.21
+// +build go1.21
+
+/*
+Copyright 2023 The logr Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package slogr
+
+import (
+	"log/slog"
+
+	"github.com/go-logr/logr"
+)
+
+// New returns a logr.Logger which writes to the same handler as the slog.Logger.
+//
+// In the output the logr verbosity level gets negated, so V(4) becomes
+// slog.LevelDebug.
+func New(logger *slog.Logger) logr.Logger {
```

Comment:
This `New` is consistent with `funcr.New`. If it wasn't for that, I probably would have used `NewLogrLogger` for consistency within this package.

Reply:
I like that better because it doesn't prefer one call over the other by giving it the shorter name. Changed.